plugins/discovery-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeServiceImpl.java

```diff
@@ -97,16 +89,62 @@ public HostedServiceGetDetailedResponse getServiceDetails() {
         }
     }
 
+    // This method is never called implicitly!
```

Comment:
If I understand the code correctly, you are not following standard lifecycle for this component. So, maybe it would make sense not to use `AbstractLifecycleComponent` here and instead just have an `initialize` method that would perform initialization. I found these comments are a bit confusing.